app/repository/pg/user.go

```diff
@@ -64,3 +63,60 @@ func (r Repo) FindUser(id string) (*entities.User, error) {
 
 	return &user, nil
 }
+
+func setDefaultIfValueIsZero(name string, value any, defaultValue any) sql.NamedArg {
```

Comment:
You don't need this.
I left a comment about QueryParams above.
As an option, you can make a method on it that will return a WHERE string of it.
After that, here in repo you just need to call the method and concat it with main SQL in a right place.

That method should add filter/sort conditions if incoming values aren't empty.

Reply:
I tried to fix `SQL` code differently. Please check it out